### ISSUE TITLE
missing title for accessibility

### ISSUE THREAD

**Issue by vellip** [django-ckeditor/django-ckeditor#424]
Right now, django-ckeditor doesn't specify a [`editor.title`](https://docs.ckeditor.com/#!/api/CKEDITOR.config-cfg-title) in the config. This results in technical ids being read out by screenreaders because it defaults to `editor.name` which basically is the initial textarea's `id` attribute.

```html
<span id="cke_id_project-result_arialbl" class="cke_voice_label">WYSIWYG-Editor, id_project-result</span>
```

is what then gets generated by CKEditor and what will be announced by the screenreader. What's " id_project-result" here, should be a human readable and descriptive name of the form field. 
This could either be something passed to `RichTextUploadingField` or simply the `verbose_name`.

**Comment by matthiask** [comment]
Our default config is here: 
https://github.com/django-ckeditor/django-ckeditor/blob/master/ckeditor/widgets.py#L34

We generally simply pass along the configuration defined by developers, and mostly do not modify it. Does it work if you add a `title` to the `CKEDITOR_CONFIGS['default']` setting in your own project?(assuming you're overriding the CKEditor configuration at all?). If yes, we should maybe just set a default title (something like "CKEditor" or "rich text editor") and advertise the possibility to override this in the documentation.

**Comment by vellip** [comment]
It would definitely be a step towards the solution. 
But I'd say often this is too generic. Since this is the text that gets read out by the SR when focussing the field, it should announce what input expected of the user. Basically like a typical `label` element's text. "Rich text Editor" would tell about the type of field, but not what is supposed to be entered in it. 
And in our case we have one custom CKEditor config that's used for all our RTE fields ("Information", "Result", "Newsletter body", etc.). So we'd have to make one custom config per field just so we can have an accessible title, which isn't really viable to us.

**Comment by xi** [comment]
> So we'd have to make one custom config per field just so we can have an accessible title, which isn't really viable to us

So this is really about allowing developers to pass a custom config to `RichTextField` instead of just a config name. This way it would be easy to overwrite the title per field.

@matthiask maybe I do not get the intended usage of `title`: Should it describe the CKEditor in general or this specific field?

**Comment by matthiask** [comment]
@xi The CKEditor docs have a better answer to this question than I: https://docs.ckeditor.com/#!/api/CKEDITOR.config-cfg-title

**Comment by matthiask** [comment]
Closing because of inactivity. Please reopen with additional details if the problem still exists.
